openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -310,7 +315,8 @@ def process_request(self, request):
             user_id = self.get_user_id_from_session(request)
             if safe_cookie_data.verify(user_id):  # Step 4
                 request.safe_cookie_verified_user_id = user_id  # Step 5
-                if LOG_REQUEST_USER_CHANGES:
+                request.safe_cookie_session_id = safe_cookie_data.session_id
+                if cache.get(USER_MISMATCH_CACHE_KEY, False) and LOG_REQUEST_USER_CHANGES:
```

Comment:
1. We need to decide if we will be using New Relic to detect issues only, or adding a Splunk alert. If we are happy with New Relic only, then we can probably solve for the missing blip problem if it actually occurs (i.e. we get a New Relic alert, but find no logs in Splunk).
2. Here is a potential comment for this line:
```
# Log useful debugging info from the request after having detected any recent mismatches
#   during any response. This will result in a missing log for the very first mismatch, but
#   hopefully the error will come in batches with logs to debug. If not, we'll need to enhance.
```